packages/next/src/server/base-server.ts

```diff
@@ -2801,6 +2801,7 @@ export default abstract class Server<
         isOnDemandRevalidate,
         isPrefetch: req.headers.purpose === 'prefetch',
         isRoutePPREnabled,
+        status: res.statusCode,
```

Comment:
This doesn't seem like where the `statusCode` should be being derived from, it seems we're failing to pass this to the latest cache entry or failing to invalidate a previous cache somewhere 🤔